benchmark/result/result.py

```diff
@@ -0,0 +1,48 @@
+"""Basic result classes for benchmarking."""
+
+from typing import Any
+from uuid import UUID
+
+from attrs import define, field
+from attrs.validators import instance_of
+from pandas import DataFrame
+
+
+@define(frozen=True)
+class Result:
+    """A single result of the benchmarking."""
+
+    name: str = field(validator=instance_of(str))
+    """The name of the benchmarking result."""
+
+    identifier: UUID = field(validator=instance_of(UUID))
+    """The unique identifier of the benchmark running which can be set
+    to compare different executions of the same benchmark setting."""
+
+    metadata: dict[str, str] = field()
```

Comment:
Would it make sense to assume/allow a certain set of metadata fields?

Reply:
I would say that it may make sense to enforce some information that helps reconstruct runs. I can think of date and time as important factors so that you can check how the benchmarks were defined when an observed run was executed.

Reply:
Well, your own benchmark does not yet provide this information ;) You'd currently save the number of doe_iterations, the batch size and the number of mc_iterations.
This raises two questions, please elaborate on them a bit and let's discuss them:
1. Should these three fields be mandatory for a `Result` object? Will they be relevant down the line in the analysis? My feeling is no, since a benchmark is assumed to never change, and the information about these things is part of the corresponding benchmark, and I do not think that the `Result` object needs this information.
2. More importantly: What is the metadata that the `Result` actually needs? Is this stuff like day of execution, used BayBE Version or similar? What is your intended use of this metadata, and how should it relate to the metadata of a benchmark?

Reply:
It's on my list for upcoming implementations, actually. The basic idea was to have a structure that can hold information that is relevant to the benchmark but doesn't fit a fixed structure. I've wanted to add further information that describes settings and parameters of the benchmark without having to open both at the same time, if that makes sense. However, if you think that is not necessary, we can also remove it. Regarding the version and date: That is a part of persisting the data from my point of view since I believe that one would separate the different benchmark results based on the version, the date when the benchmark was started (since I guess most people would search for the start-date of the execution; it should be the same for all results from this particular run), the branch where it has been started, the workflow where it ran, and so on. This makes them a good combined key candidate for something like a file system. If that information should be part of the metadata (which makes sense now that I think about it), they should be mandatory for further use in the planned persistence component. But if so, we must think about different environments where the benchmark may be executed with also different mandatory information (locally executed benchmarks don't have a pipeline workflow ID, but this information might be relevant if results from the pipeline are observed for backtracking purposes). But that must be part of another PR. Especially which environments are relevant has to be defined, from my point of view. Long story short: I wanted to keep that layer of complexity from the result in general and don't use them for benchmark separation but for a container of additional, unrestricted information provided by the callable or result itself.

Sorry for the long comment. I was unsure how to put this, which is why I included further but currently irrelevant information.

Reply:
I guess that this can now be resolved since things have changed a bit with the explicit introduction of the metadata class, right? Ir is there anything we should discuss here? If not, feel free to resolve.